fmriprep/interfaces/freesurfer.py

```diff
@@ -68,3 +75,99 @@ def cmdline(self):
+            self.inputs.subject_id,
+            self.inputs.in_brain)
+        self._results['subjects_dir'] = subjects_dir
+        self._results['subject_id'] = subject_id
+        return runtime
+
+
+class FSDetectInputsInputSpec(BaseInterfaceInputSpec):
+    t1w_list = InputMultiPath(File(exists=True), mandatory=True,
+                              desc='input file, part of a BIDS tree')
+    t2w_list = InputMultiPath(File(exists=True), desc='input file, part of a BIDS tree')
+    hires_enabled = traits.Bool(True, usedefault=True, desc='enable hi-resolution processing')
+
+
+class FSDetectInputsOutputSpec(TraitedSpec):
+    t2w = File(desc='reference T2w image')
+    use_t2w = traits.Bool(desc='enable use of T2w downstream computation')
+    hires = traits.Bool(desc='enable hi-res processing')
+    mris_inflate = traits.Str(desc='mris_inflate argument')
+
+
+class FSDetectInputs(SimpleInterface):
+    input_spec = FSDetectInputsInputSpec
+    output_spec = FSDetectInputsOutputSpec
+
+    def _run_interface(self, runtime):
+        t2w, self._results['hires'], mris_inflate = detect_inputs(
+            self.inputs.t1w_list,
+            hires_enabled=self.inputs.hires_enabled,
+            t2w_list=self.inputs.t2w_list if isdefined(self.inputs.t2w_list) else None)
+
+        self._results['use_t2w'] = t2w is not None
+        if self._results['use_t2w']:
+            self._results['t2w'] = t2w
+
+        if self._results['hires']:
+            self._results['mris_inflate'] = mris_inflate
+
+        return runtime
+
+
+def inject_skullstripped(subjects_dir, subject_id, skullstripped):
+    mridir = op.join(subjects_dir, subject_id, 'mri')
+    t1 = op.join(mridir, 'T1.mgz')
+    bm_auto = op.join(mridir, 'brainmask.auto.mgz')
+    bm = op.join(mridir, 'brainmask.mgz')
+
+    if not op.exists(bm_auto):
+        img = nb.load(t1)
+        mask = nb.load(skullstripped)
+        bmask = new_img_like(mask, mask.get_data() > 0)
+        resampled_mask = resample_to_img(bmask, img, 'nearest')
+        masked_image = new_img_like(img, img.get_data() * resampled_mask.get_data())
+        masked_image.to_filename(bm_auto)
+
+    if not op.exists(bm):
+        copyfile(bm_auto, bm, copy=True, use_hardlink=True)
+
+    return subjects_dir, subject_id
+
+
+def detect_inputs(t1w_list, t2w_list=None, hires_enabled=True):
+    t1w_list = filename_to_list(t1w_list)
+    t2w_list = filename_to_list(t2w_list) if t2w_list is not None else []
+    t1w_ref = nb.load(t1w_list[0])
+    # Use high resolution preprocessing if voxel size < 1.0mm
+    # Tolerance of 0.05mm requires that rounds down to 0.9mm or lower
+    hires = hires_enabled and max(t1w_ref.header.get_zooms()) < 1 - 0.05
+
+    t2w = None
+    if t2w_list and max(nb.load(t2w_list[0]).header.get_zooms()) < 1.2:
+        t2w = t2w_list[0]
+
+    # https://surfer.nmr.mgh.harvard.edu/fswiki/SubmillimeterRecon
+    mris_inflate = '-n 50' if hires else None
+    return (t2w, hires, mris_inflate)
```

Comment:
@effigies still considering to move the code into the interfaces? Otherwise, this is ready to merge.